### ISSUE TITLE
Case Insensitivity and Partial Words in Search

### ISSUE THREAD

**Issue by yorrd** [silverbulletmd/silverbullet#219]
The current global search feature only looks for exact matches. In everyday use, I'm accustomed to case-insensitivity and being able to search for partial words (if there is "SilverBullet" in any file, I'd like to find it by searching for "silver" or "bullet".

**Comment by zefhemel** [comment]
Yeah I'm not surprised. The whole full text search thing needs to be properly looked and may have to be redone. What's there now is the least effort thing to get something working, I'm sure it can use a lot of tweaking.

**Comment by itsamemarkus** [comment]
Regarding the search function, links have the highlight syntax and are not clickable

---

![image](https://user-images.githubusercontent.com/34944988/216723780-7171c5e9-8865-4618-8d05-503c1dc5e81b.png)

---

![image](https://user-images.githubusercontent.com/34944988/216723852-012d96df-a943-418f-91d8-9b66aaed33ec.png)

**Comment by v411e** [comment]
Do you prefer to have minimal dependencies? If not, we could think about using a dedicated library like https://github.com/krisk/Fuse
